pkg/agent/consistenthash/consistenthash.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package consistenthash provides an implementation of a ring hash.
+package consistenthash
```

Comment:
Wonder are we copying code/ideas from any other projects? If so , probably we should add a note about that.

Reply:
Added the source link and some notes. Thanks!